learn/features/randomness.md

```diff
@@ -0,0 +1,95 @@
+
+When randomness is requested through the precompile's `requestLocalVRFRandomWords` or `requestRelayBabeEpochRandomWords` method, a fee is set aside to pay for the fulfillment of the request. When using local VRF, to increase unpredictability, a specified delay period in blocks, must pass before the request can be fulfilled. At the very least, the delay period must be greater than one block. For BABE epoch randomness, you do not need to specify a delay but can fulfill the request at the beginning of the 2nd epoch following the current one.
+
+After the delay, fulfillment of the request can be manually executed by anyone through the [`fulfillRequest`](/builders/pallets-precompiles/precompiles/randomness/#:~:text=fulfillRequest){target=_blank} method using the fee that was initially set aside for the request.
+
+When fulfilling the randomness request via the precompile's `fulfillRequest` method, the [`rawFulfillRandomWords`](/builders/pallets-precompiles/precompiles/randomness/#:~:text=rawFulfillRandomWords(uint256 requestId, uint256[] memory randomWords)){target=_blank} function in the `RandomnessConsumer.sol` contract will be called, which will verify that the sender is the randomness precompile. From there, [`fulfillRandomWords`](/builders/pallets-precompiles/precompiles/randomness/#:~:text=fulfillRandomWords(uint256 requestId, uint256[] memory randomWords)){target=_blank} is called and the requested number of random words are computed using the current block's randomness result and a given salt and returned. If the fulfillment was successful, the [`FulfillmentSucceeded` event](/builders/pallets-precompiles/precompiles/randomness/#:~:text=FulfillmentSucceeded){target=_blank} will be emitted; otherwise the [`FulfillmentFailed` event](/builders/pallets-precompiles/precompiles/randomness/#:~:text=FulfillmentFailed){target=_blank} will be emitted. 
+
+For fulfilled requests, the cost of execution will be refunded from the request fee to the caller of `fulfillRequest`. Then any excess fees and the request deposit are transferred to the specified refund address.
+
+Your contract's `fulfillRandomWords` callback is responsible for handling the fulfillment. For example, in a lottery contract, the callback would use the random words to choose a winner and payout the winnings.
+
+If a request expires it can be purged through the precompile's [`purgeExpiredRequest` function](/buildxers/pallets-precompiles/precompiles/randomness/#:~:text=purgeExpiredRequest){target=_blank}. When this function is called the request fee is paid out to the caller and the deposit will be returned to the original requester.
+
+The happy path for a randomness request is shown in the following diagram:
+
+![Randomness request happy path diagram](/images/learn/features/randomness/randomness-1.png)
```

Comment:
Loved the image!!! - I'm a bit confused tho, both refund Fees and Deposit goes to a different person? My main comment is that I would clarify in the image that the fees can go to one person and the deposit can go to another (specified in the request right)?

Reply:
for fulfilled requests (happy path) the cost of execution can go to the caller of the fulfillment and then the excess fees and deposit are transferred to the refund address -- updated the diagram to show this